### PR TITLE
Feature/wizard WIP

### PR DESCRIPTION
**Pull request by jasperswart**
## React Boilerplate

Thank you for contributing! Please take a moment to review our [**contributing guidelines**](https://github.com/react-boilerplate/react-boilerplate/blob/master/.github/CONTRIBUTING.md)
to make the process easy and effective for everyone involved.

**Please open an issue** before embarking on any significant pull request, especially those that
add a new library or change existing tests, otherwise you risk spending a lot of time working
on something that might not end up being merged into the project.

Before opening a pull request, please ensure:

- [ ] You have followed our [**contributing guidelines**](https://github.com/react-boilerplate/react-boilerplate/blob/master/.github/CONTRIBUTING.md)
- [ ] double-check your branch is based on `dev` and targets `dev` 
- [ ] Pull request has tests (we are going for 100% coverage!)
- [ ] Code is well-commented, linted and follows project conventions
- [ ] Documentation is updated (if necessary)
- [ ] Internal code generators and templates are updated (if necessary)
- [ ] Description explains the issue/use-case resolved and auto-closes related issues

Be kind to code reviewers, please try to keep pull requests as small and focused as possible :)

**IMPORTANT**: By submitting a patch, you agree to allow the project
owners to license your work under the terms of the [MIT License](https://github.com/react-boilerplate/react-boilerplate/blob/master/LICENSE.md).

### CODE REVIEW

File: app/meldingen/incident/containers/IncidentContainer/saga.js [hunk 1]
```diff
@@ -1,6 +1,49 @@
...
+  getClassificationError
+} from './actions';
+import makeSelectIncidentContainer from './selectors';
+
+export function* createIncident() {
+  const requestURL = '/api/meldingen';
+
+  try {
+    const data = yield select(makeSelectIncidentContainer());
+    console.log('data', data);
```
Comment (aditudorache) [inline app/meldingen/incident/containers/IncidentContainer/saga.js]: no console.log

File: app/meldingen/incident/containers/IncidentContainer/saga.js [hunk 2]
```diff
@@ -1,6 +1,49 @@
...
+    const incident = yield call(request, requestURL);
+    yield put(createIncidentSuccess(incident));
+  } catch (err) {
+    yield put(createIncidentError(err));
+  }
+}
+
+export function* getClassification({ text }) {
+  const requestURL = 'http://meldingen-classification.herokuapp.com/calls/';
```
Comment (aditudorache) [inline app/meldingen/incident/containers/IncidentContainer/saga.js]: use this through the dyson middleware.
const requestURL="<extern-api-name>/getclasification";

in proxy.conf.dev.js you can add the entry:
'/<extern-api-name>/*': {
      target: 'http://meldingen-classification.herokuapp.com/calls/',
      pathRewrite: { '^/<extern-api-name>/': '' },
      secure: false,
      changeOrigin: true,
      logLevel: 'debug'
    }

File: app/meldingen/incident/containers/IncidentContainer/saga.js [hunk 3]
```diff
@@ -1,6 +1,49 @@
...
+    yield put(getClassificationError(err));
+  }
+}
 
 // Individual exports for testing
-export default function* defaultSaga() {
-  // See example in containers/HomePage/saga.js
+export default function* watchIncidentContainerSaga() {
+  yield takeLatest(CREATE_INCIDENT, createIncident);
+  yield takeLatest(GET_CLASSIFICATION, getClassification);
```
Comment (aditudorache) [inline app/meldingen/incident/containers/IncidentContainer/saga.js]: the two yields are waiting for each other. use 
yield [
takeLatest(...),
takeLatest(...)
]

to start them in parallel

File: app/meldingen/incident/definitions/wizard.js [hunk 4]
```diff
@@ -1,19 +1,58 @@
...
+        },
+        $field_0: {
+          isStatic: false,
+          render: IncidentNavigation
+        }
       }
-    ]
+    }
   },
   // vulaan: {
```
Comment (aditudorache) [inline app/meldingen/incident/definitions/wizard.js]: comments

File: app/meldingen/incident/components/IncidentForm/components/DescriptionWithClassificationInput/index.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import ErrorMessage from '../ErrorMessage/';
+
+const DescriptionWithClassificationInput = ({ handler, touched, hasError, meta, parent }) => (
+  <div>
+    <textarea
+      placeholder={meta.placeholder}
+      onKeyUp={(e) => parent.meta.getClassification(e.target.value)}
```
Comment (vjawala) [inline app/meldingen/incident/components/IncidentForm/components/DescriptionWithClassificationInput/index.js]: parent.meta? what does this do? Can't you just pass along this function directly as a prop?

File: app/meldingen/incident/components/IncidentForm/index.js [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+      getClassification: this.props.getClassification,
+      setIncident: this.props.setIncident
+    };
+
+    this.setValues(incident);
+  }
+
+  setValues(incident, onlyWatchedItems = false) {
+    if (this.form && this.form.controls) {
+      window.setTimeout(() => {
```
Comment (vjawala) [inline app/meldingen/incident/components/IncidentForm/index.js]: Kan dit niet in de componentDidMount?

File: app/meldingen/incident/components/IncidentForm/style.scss [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+    background-color: white;
+    padding: 2px 8px;
+
+    &:read-only {
+      color: grey;
+    }
+  }
+
+  .field-errors {
+    color: red;
```
Comment (vjawala) [inline app/meldingen/incident/components/IncidentForm/style.scss]: try to keep to the ams-colorpallette variables

File: app/meldingen/incident/components/IncidentPreview/components/PlainText/index.js [hunk 8]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const Text = ({ text }) => (
```
Comment (vjawala) [inline app/meldingen/incident/components/IncidentPreview/components/PlainText/index.js]: Is this the style for the props we're using throughout the entire app? Or do we want to pass it as `(props)` and use it as `props.text`? I have no preference, but I think we should be sure that one style is used

File: app/meldingen/incident/components/IncidentWizard/index.js [hunk 9]
```diff
@@ -14,44 +14,52 @@ import { Wizard, Steps, Step } from 'react-albus';
...
-          </div>
-        </div>
+        <Route
+          render={({ history }) => (
+            <Wizard history={history}>
+              <Steps>
+                {Object.keys(wizard).map((key) => (
+                  <Step key={key} id={`incident/${key}`}>
+                    <h1 className="text-align-center">{key}</h1>
+                    {wizard[key].preview ?
```
Comment (vjawala) [inline app/meldingen/incident/components/IncidentWizard/index.js]: `Boolean(wizard[key].preview) && <Incide...`

File: app/meldingen/incident/components/IncidentWizard/index.js [hunk 10]
```diff
@@ -14,44 +14,52 @@ import { Wizard, Steps, Step } from 'react-albus';
...
+                      <IncidentPreview
+                        incident={incident}
+                        preview={wizard[key].preview}
+                      />
+                      : ''}
+
+                    {wizard[key].form ?
```
Comment (vjawala) [inline app/meldingen/incident/components/IncidentWizard/index.js]: ^

File: app/meldingen/incident/containers/IncidentContainer/actions.js [hunk 11]
```diff
@@ -5,11 +5,60 @@
...
+  CREATE_INCIDENT,
+  CREATE_INCIDENT_SUCCESS,
+  CREATE_INCIDENT_ERROR,
+  GET_CLASSIFICATION,
+  GET_CLASSIFICATION_SUCCESS,
+  GET_CLASSIFICATION_ERROR
 } from './constants';
 
-export function defaultAction() {
+export function setIncident(incident) {
```
Comment (vjawala) [inline app/meldingen/incident/containers/IncidentContainer/actions.js]: `export function setIncident()` or `export const setIncident = () =>`?
Same goes every export in the app, we have to keep this uniform

**Pull request merged**